crates/shared/src/signature_validator/simulation.rs

```diff
@@ -131,7 +121,7 @@ impl Simulation {
     fn decode(output: &[u8]) -> Result<Self> {
         let function = contracts::support::Signatures::raw_contract()
             .abi
-            .function("balance")
+            .function("validate")
```

Comment:
Did this never cause problems?

Reply:
It did - the PR this was based off of was untested. I noticed this while implementing #1832